src/lib/__tests__/routing-test.ts

```diff
@@ -0,0 +1,152 @@
+import { createSearchClient } from '../../../test/mock/createSearchClient';
+import { wait } from '../../../test/utils/wait';
+import historyRouter from '../routers/history';
+import instantsearch from '../..';
+import { connectSearchBox } from '../../connectors';
+
+describe('routing', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('writeOnDispose=true', () => {
+    test('cleans URL on dispose', async () => {
+      const pushState = jest.spyOn(window.history, 'pushState');
+      const writeDelay = 400;
+
+      const search = instantsearch({
+        indexName: 'indexName',
+        searchClient: createSearchClient(),
+        routing: {
+          router: historyRouter({
+            writeDelay,
+          }),
+        },
+      });
+
+      search.addWidgets([connectSearchBox(() => {})({})]);
+
+      search.start();
+
+      // Wait for ${writeDelay} to pass then check URL has been initialized
+      await wait(1.5 * writeDelay);
```

Comment:
I'm assuming the 1.5 factor is to ensure the `writeDelay` did elapse and avoid false positives. Since we run this 10 times from what I've counted, we're adding 6 seconds to this test suite.

Could we reduce `writeDelay` to avoid slowing down the test suite too much?

Reply:
fake timers should also fix that

Reply:
Yep, I assumed this was tried already.

Reply:
@sarahdayan I've reduce the `writeDelay` to 10ms, which bring the total waiting time to 600ms, do you think that's acceptable?

Reply:
Did you try fake timers? Not sure they work in that situation but worth trying.

Otherwise, 600ms sounds reasonable.

Reply:
Yes, but every test exceed the max timeout, it seems like the same issue I have: https://github.com/nock/nock/issues/2200